core/posts/views.py

```diff
@@ -156,6 +158,27 @@ def retrieve(self, request, pk):
             "images": img_serializer.data
         })
 
+    @action(detail=True)
+    def image(self, request, pk):
+        try:
+            post = Posts.objects.get(pk=pk)
+        except:
+            return Response({
+                "success": False,
+                "message": "No post was found with that ID",
+                "query": "getImage"
+            }, status=404)
+        if not can_user_view(request.user, post):
```

Comment:
Does this actually work on the front-end? I'm not 100% sure that the authenticated user is included in the request since it's made from an img tag

Reply:
yes it works :)

Reply:
What if the visibility isn't public?

Reply:
I tested on local with private and foaf, shows a 403 when trying to get the image - otherwise it's shown if allowed